llvm/test/SYCLLowerIR/esimd_global.ll

```diff
@@ -0,0 +1,166 @@
+; This test checks whether globals are converted
+; correctly to llvm's native vector type.
+;
+; RUN: opt < %s -ESIMDLowerVecArg -S | FileCheck %s                            
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown-sycldevice"
+
+%"class._ZTSN2cl4sycl5intel3gpu4simdIiLi16EEE.cl::sycl::intel::gpu::simd" = type { <16 x i32> }
+
+$"_ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE4Test" = comdat any
+
+; CHECK: [[NEWGLOBAL:[@a-zA-Z0-9_]*]] = dso_local global {{.+}} align 64 #0
```

Comment:
Why are you not checking the type and the initializer here?

Reply:
fixed